### PR TITLE
Adding Output writer to containers.

### PR DESCRIPTION
**Pull request by karthiknadig**
#3765

### CODE REVIEW

File: src/Windows/Containers/Impl/Docker/WindowsDockerService.cs [hunk 1]
```diff
@@ -19,10 +20,12 @@ public class WindowsDockerService : LocalDockerService, IContainerService {
         const string DockerServiceName = "Docker for Windows";
         private readonly IFileSystem _fs;
         private readonly IProcessServices _ps;
+        private readonly IActionLogWriter _outputLogWriter;
 
-        public WindowsDockerService(IFileSystem fs, IProcessServices ps, IRegistry registryService):base (GetLocalDocker(registryService, fs), ps) {
+        public WindowsDockerService(IFileSystem fs, IProcessServices ps, IRegistry registryService, IActionLogWriter logWriter = null) : base (GetLocalDocker(registryService, fs), ps, logWriter) {
             _fs = fs;
             _ps = ps;
+            _outputLogWriter = logWriter;
```
Comment (AlexanderSher) [inline src/Windows/Containers/Impl/Docker/WindowsDockerService.cs]: Field isn't used right now - is it for the future reference?

File: src/Package/Impl/Logging/OutputWindowLog.cs [hunk 2]
```diff
@@ -43,9 +43,13 @@ internal sealed class OutputWindowLogWriter {
         }
 
         public Task WriteAsync(MessageCategory category, string message) {
+            Write(category, message);
+            return Task.CompletedTask;
+        }
+
+        public void Write(MessageCategory category, string message) {
```
Comment (AlexanderSher) [inline src/Package/Impl/Logging/OutputWindowLog.cs]: Kill that `WriteAsync` please. It is 100% sync.

File: src/Containers/Impl/Docker/LocalDockerService.cs [hunk 3]
```diff
@@ -17,13 +18,15 @@ namespace Microsoft.R.Containers.Docker {
     public abstract class LocalDockerService : IDockerService {
         private readonly IProcessServices _ps;
         private readonly LocalDocker _docker;
+        private readonly IActionLogWriter _outputLogWriter;
         private readonly Regex _containerIdMatcher64 = new Regex("[0-9a-f]{64}", RegexOptions.IgnoreCase);
         private readonly Regex _containerIdMatcher12 = new Regex("[0-9a-f]{12}", RegexOptions.IgnoreCase);
         private int _defaultTimeout = 5000;
 
-        public LocalDockerService(LocalDocker docker, IProcessServices ps) {
+        public LocalDockerService(LocalDocker docker, IProcessServices ps, IActionLogWriter logWriter) {
```
Comment (MikhailArkhipov) [inline src/Containers/Impl/Docker/LocalDockerService.cs]: What creates it? In SQL I do

```
            _outputWindow = new OutputWindowLogWriter(services, VSConstants.OutputWindowPaneGuid.BuildOutputPane_guid, string.Empty);
```

Comment (karthiknadig) [inline src/Containers/Impl/Docker/LocalDockerService.cs]: I didn't want to create the writer until the UI for this is available.

File: src/Package/Impl/Logging/OutputWindowLog.cs (hunk 2 above)
Comment (karthiknadig) [inline src/Package/Impl/Logging/OutputWindowLog.cs]: will do.

File: src/Containers/Impl/Docker/LocalDockerService.cs [hunk 4]
```diff
@@ -138,8 +141,19 @@ public abstract class LocalDockerService : IDockerService {
             string output = await process.StandardOutput.ReadToEndAsync();
             string error = await process.StandardError.ReadToEndAsync();
             if (!string.IsNullOrEmpty(error)) {
+                try {
+                    _outputLogWriter?.Write(MessageCategory.Error, error);
+                } catch(Exception ex) when (!ex.IsCriticalException()) {
```
Comment (AlexanderSher) [inline src/Containers/Impl/Docker/LocalDockerService.cs]: `logWriter` shouldn't throw any exceptions.

Comment (MikhailArkhipov) [inline src/Containers/Impl/Docker/LocalDockerService.cs]: It is a COM call in VS FWIW

Comment (AlexanderSher) [inline src/Containers/Impl/Docker/LocalDockerService.cs]: `FileLogWriter` handles all IO-related exceptions exceptions by itself (see `WriteToFileAsync`), so I think `OutputWindowLogWriter` should do the same.

**Comment by karthiknadig** [comment]
Added several minor changes to this PR.

File: src/Windows/Setup/RemoteServices/Product.wxs [hunk 5]
```diff
@@ -43,6 +43,7 @@
     <DirectoryRef Id="INSTALLFOLDER">
       <Component Id="ProductAssemblies" Guid="{9C717F8F-4A97-4FA5-8031-5A180467754D}" Win64="yes">
         <?include "Assemblies.wxi" ?>
+        <File Source="$(var.BinDir)\Microsoft.R.Host.Broker.dll" />
```
Comment (AlexanderSher) [inline src/Windows/Setup/RemoteServices/Product.wxs]: I think it should be in `Assemblies.wxi`

**Comment by MikhailArkhipov** [comment]
👍

**Pull request merged**